packages/react/src/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -223,27 +243,64 @@ const ToggleSwitch = React.forwardRef<HTMLButtonElement, React.PropsWithChildren
       onClick,
       size = 'medium',
       statusLabelPosition = 'start',
+      loadingLabelDelay = 2000,
```

Comment:
Could you share more about what folks would use the delay here for? Wanted to learn more 👀 I assume a delay is nice in case it resolves quickly so we didn't need to announce anything?

Reply:
Definitely! [We received feedback some time ago on loading states](https://github.com/github/primer/issues/2520#issuecomment-1681405810), and it was mentioned that there may be cases where users quickly toggle between on/off and the loading state is quick enough so that a loading message wouldn't be needed (e.g. < 2 seconds).

The time we've added (2 seconds) is a bit arbitrary, but was a suggestion on how long we _could_ wait before adding the loading message.